presto-hive/src/main/java/com/facebook/presto/hive/HiveCompressionCodec.java

```diff
@@ -24,13 +24,15 @@
 
 import static com.facebook.presto.hive.HiveStorageFormat.DWRF;
 import static com.facebook.presto.hive.HiveStorageFormat.ORC;
+import static com.facebook.presto.hive.HiveStorageFormat.PAGEFILE;
 import static java.util.Objects.requireNonNull;
 
 public enum HiveCompressionCodec
 {
     NONE(null, CompressionKind.NONE, CompressionCodecName.UNCOMPRESSED, f -> true),
     SNAPPY(SnappyCodec.class, CompressionKind.SNAPPY, CompressionCodecName.SNAPPY, f -> true),
-    GZIP(GzipCodec.class, CompressionKind.ZLIB, CompressionCodecName.GZIP, f -> true),
+    LZ4(null, CompressionKind.NONE, null, f -> f == PAGEFILE),
```

Comment:
Why do we need another compression? ZSTD with lower levels should be able to achieve comparable speed with much better compression.

Reply:
We use LZ4 for shuffle. Because we care performance more rather than storage saving. The temp tables are short-living.

Reply:
How about not using any compression?

Reply:
That is also considered in our exchange client buffer; if after compression we find the size is larger than the original bytes, we will send the original bytes. The flag is located in the header of a serialized page. Check `PagesSerde`.

Reply:
> We use LZ4 for shuffle. Because we care performance more rather than storage saving. The temp tables are short-living.

Right, that's why I added LZ4 option. I am thinking about adding ZSTD too, but I haven't done any benchmark on real queries yet. In my impression, SNAPPY and LZ4 are faster.

Reply:
SNAPPY and LZ4 are indeed faster for text data. But I think with the combination of no compression + ZSTD level 1 should be good enough. Needs to be experimented for our data.

Reply:
SNAPPY and LZ4 are faster, but provide lower level of compression. My question is why do we need both, `LZ4` and `SNAPPY`? `SNAPPY` and `LZ4` are very similar characteristics wise. I wonder if it makes sense to leave only `SNAPPY` for now, as it already exist and supported by all storage formats.